### PR TITLE
GCE: ILB E2E test

### PR DESCRIPTION
**Pull request by nicksardo**
/assign @bowei 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/service.go [hunk 1]
```diff
@@ -1386,26 +1404,35 @@ var _ = framework.KubeDescribe("Services", func() {
...
-		}); pollErr != nil {
-			framework.Failf("Loadbalancer IP not changed to internal.")
+			// should have the given static internal IP.
+			jig.SanityCheckService(svc, v1.ServiceTypeLoadBalancer)
+			Expect(framework.GetIngressPoint(lbIngress)).To(Equal(internalStaticIP))
+		}
+
+		if framework.ProviderIs("gce", "gke") {
+			By("switching to ClusterIP type to destroy loadbalancer")
+			jig.ChangeServiceType(svc.Namespace, svc.Name, v1.ServiceTypeClusterIP, createTimeout)
```
Comment (nicksardo) [inline test/e2e/service.go]: Could remove this cloud-specific check. Didn't want to make functional changes to azure without permission from somebody.

File: test/e2e/service.go [hunk 2]
```diff
@@ -1337,39 +1337,57 @@ var _ = framework.KubeDescribe("Services", func() {
 		framework.CheckReachabilityFromPod(true, normalReachabilityTimeout, namespace, dropPodName, svcIP)
 	})
 
-	It("should be able to create an internal type load balancer on Azure [Slow]", func() {
-		framework.SkipUnlessProviderIs("azure")
+	It("should be able to create an internal type load balancer [Slow]", func() {
```
Comment (bowei) [inline test/e2e/service.go]: I would prefer to just split this into two different tests if we are going to have so much provider specific logic

Comment (nicksardo) [inline test/e2e/service.go]: I moved the ILB-enabling/disabling portion of the test to framework. The only cloud-specific portion left is the internal static IP for azure which I believe will be removed in a few weeks after adding support for GCE.

**Pull request merged**